### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by mbaudin47**
This PR improves several examples.

- The [plot-kriging-chose-trend-py](http://openturns.github.io/openturns/master/auto_meta_modeling/kriging_metamodel/plot_kriging_chose_trend.html#sphx-glr-auto-meta-modeling-kriging-metamodel-plot-kriging-chose-trend-py) has issues :

image

- The [EnumerateFunction](http://openturns.github.io/openturns/master/user_manual/_generated/openturns.EnumerateFunction.html#openturns.EnumerateFunction) is unclear.
- The `vect` macro is not always used.

### CODE REVIEW

**Review by jschueller — APPROVED** [review]
a rebase is needed
any thoughts @josephmure ?

**Comment by mbaudin47** [comment]
@m-balesdent : I would highly appreciate your review if you are interested on this topic.

**Comment by JPelamatti** [comment]
Everything looks good to me, except for the model being referred as `g()` in the notation section. I believe there is no consensus on the notation between different communities, and even so in our documentation. A preliminary discussion is required in my opinion.

**Comment by jschueller** [comment]
it probably needs a rebase

**Comment by m-balesdent** [comment]
Thanks @mbaudin47 for this work.

Some improvements that can be done to enhance this example follow.

**Comment by m-balesdent** [comment]
The links to ConstantBasis, LinearBasis and QuadraticBasis seem to be broken on the sphinx doc.

**Comment by m-balesdent** [comment]
image

I think it should be "Gaussian".

**Comment by m-balesdent** [comment]
The link to `ParametricFunction` is broken 

image

**Comment by m-balesdent** [comment]
Maybe it should be in mathmode ?

image

**Comment by m-balesdent** [comment]
The closing parenthesis is not in the same line in this sentence : 

image

**Comment by m-balesdent** [comment]
I'm not sure to understand the use of uppercase letters in this sentence because we do not deal with aleatory variables.

image

**Comment by m-balesdent** [comment]
I think that this kind of figures are quite confusing : 

image

Usually, when we plot a confidence interval, we expect to see the confidence interval of the kriging model (that collapses in the data points).  Here, it is not. It is the scale parameter applied on the trend. I think at least a sentence could be added to explain this point.

 A better thing could be to plot both confidence interval on the trend and on the kriging model.

**Comment by m-balesdent** [comment]
There is a problem with the tag `sphinx_gallery_thumbnail_number`

image

**Comment by m-balesdent** [comment]
I think there is a problem also at the end of the example. Nothing appears after the `plt.show()` command.

image


That's all for me. Thanks for this PR.

**Comment by mbaudin47** [comment]
> I think that this kind of figures are quite confusing :
> [...]
> A better thing could be to plot both confidence interval on the trend and on the kriging model.

I agree. It appears that @KieranDelamotte  and @josephmure created that particular example at a1e667923fda3dc39d3184dbe3e33dcf06f3e93a. I think that the confidence interval are wrong. In my opinion, they should look like the one presented in [plot_kriging_1d](http://openturns.github.io/openturns/latest/auto_meta_modeling/kriging_metamodel/plot_kriging_1d.html#compute-confidence-bounds). Do you agree @KieranDelamotte?

**Comment by mbaudin47** [comment]
OK, thank your for that answer. So I have to work on this example before it can be considered for merging again.

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py [hunk 1]
```diff
@@ -10,94 +10,106 @@
...
 #
-# In this example we present the basic trends which we may use in the kriging metamodel procedure :
+# In this example we present the trends which we may use in a kriging metamodel:
 #
-#  - ConstantBasis ;
-#  - LinearBasis ;
-#  - QuadraticBasis ;
+# - :class:`~openturns.ConstantBasisFactory`;
+# - :class:`~openturns.LinearBasisFactory`;
+# - :class:`~openturns.QuadraticBasisFactory`.
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py:17]: Fixed at bb88a3c87bf269486726a84967f9440ebb6f6c21

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 """
-Kriging : choose a trend vector space
+Kriging: choose a trend vector space
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py:2]: Fixed at bb88a3c87bf269486726a84967f9440ebb6f6c21

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py [hunk 3]
```diff
@@ -316,29 +340,21 @@ def plot_ICBounds(x_test, y_test, nTest, sigma):
...
-graph.setLegendPosition("bottom")
-graph.setTitle("1D Kriging : exact model and metamodel")
-view = otv.View(graph)
+graph = plot_exact_model(palette[0])
+graph.add(plotMetamodel(x_test, result, myTransform, palette[1]))
 
 
 # %%
-#  We can retrieve the calibrated trend coefficients with `getTrendCoefficients` :
+#  We can retrieve the calibrated trend coefficients with :meth:`KrigingResult.getTrendCoefficients` :
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py:355]: Fixed at bb88a3c87bf269486726a84967f9440ebb6f6c21

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py [hunk 4]
```diff
@@ -351,69 +367,31 @@ def plot_ICBounds(x_test, y_test, nTest, sigma):
...
-# still have to impose the interpolation property.
-#
-# We see that the gap between the trend and the exact model is small and the amplitude of the gaussian
-# process is small as well : :math:`\sigma \approx 0.684`. Then the confidence interval is more
-# narrow than before. The scale parameter is small, :math:`\rho \approx 0.010`, meaning that two
-# distant points are rapidly independent. The only way to set the interpolation property is to create
-# small peaks (see the figure). The sad conclusion is that the metamodel loses its smoothness.
-#
-# In that case one should use a constant or linear basis even if a quadratic trend seems right.
+# sphinx_gallery_thumbnail_number = 10
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_chose_trend.py:384]: Fixed at bb88a3c87bf269486726a84967f9440ebb6f6c21

**Comment by mbaudin47** [comment]
@m-balesdent , @JPelamatti  : May I suggest an updated review?

**Review by sofianehaddad — APPROVED** [review]

**Comment by mbaudin47** [comment]
@josephmure : The PR seems ready for merging as far as I can see. May I suggest a review?

**Comment by mbaudin47** [comment]
@jschueller : Given that we have 2 positive reviews and non-respondent reviewers without any negative review, I suggest to merge.

**Pull request merged**